scom/stackstate_checks/scom/scom.py

```diff
@@ -157,6 +185,13 @@ def check(self, instance):
                 root_ids = []
                 root_tree = response.get("rows", [])
                 for component in root_tree:
+                    type_data = {
+                            "criteria": "(Id = '"+component.get("id")+"')"
+                            }
+                    type_response = (session.post('http://'+ scom_ip +'/OperationsManager/data/scomObjects',
+                                auth=HttpNtlmAuth(domain + '\\' + username, password),
+                                json=type_data["criteria"])).json()
+                    component_type = type_response.get("scopeDatas")[0].get("className").lower().replace(" ","-")
```

Comment:
what is this duplicate code?

Reply:
this was not complete, i have rewritten the whole integration and combined multiple operations together to save some requests.